### ISSUE TITLE
Array.reduce not trustworthy

### ISSUE THREAD

**Issue by eoghanmurray** [rrweb-io/rrweb-snapshot#51]
After a lot of debugging I traced an rrweb error thrown to a case where a 3rd party library on a website was defining it's own version of `Array.reduce`

We've no control over what other js is run on the website, for the record the code was:
```
iWeb - iWebSite.js
//  Copyright (c) 2007-2008 Apple Inc. All rights reserved.

//  This file includes a copy of the Prototype JavaScript framework:
```

Is this something worth fixing, e.g. by 'unrolling' the reduce call into an accumulator loop?

https://github.com/rrweb-io/rrweb-snapshot/blob/1251091d6446cf8fe360012ff4b0a4af98567aff/src/snapshot.ts#L38

**Comment by eoghanmurray** [comment]
Have added a fix in https://github.com/rrweb-io/rrweb-snapshot/pull/52

**Comment by Yuyz0112** [comment]
I've merged #52 

I think I've debugged some similar problems before. It was a little hard to defend all of this kind of issue, but it's ok to do some try-my-best fix.

**Comment by eoghanmurray** [comment]
Ya, fault is not with rrweb
